docs/UserGuide.md

```diff
@@ -616,6 +617,10 @@ Examples:
 * `range deleteperson from/2 to/3`
   * Deletes the 2nd and 3rd contacts in the address book.
 
+<div markdown="span" class="alert alert-warning">:rotating_light: **Important Note:**
+The `batch` `range` operation are transactions, it will fail due to one command executed failed and the error message of that error command will show, no data will modify or save. 
```

Comment:
Separate the warning to individually batch and range (can copy paste).
```suggestion
The `range` operation is transactional, either all changes succeed or all changes fail (should an operation within `range` fails). Any error message shown is from the failing operation.
```

Reply:
thx changed